subsys/bluetooth/controller/ticker/ticker.h

```diff
@@ -101,6 +109,26 @@ typedef void (*ticker_timeout_func) (uint32_t ticks_at_expire,
 				     uint16_t lazy, uint8_t force,
 				     void *context);
 
+
+struct ticker_expire_info {
+	uint32_t ticks_to_expire;
+	uint32_t remainder;
+	uint16_t lazy;
+};
+
+struct ticker_ext_timeout_data {
+	uint32_t ticks_at_expire;
+	uint32_t ticks_drift;
+	uint32_t remainder;
+	uint16_t lazy;
+	uint8_t force;
+	struct ticker_expire_info *other_expire_info;
+};
+
+/** \brief Timer extended timeout function type.
+ */
+typedef void (*ticker_ext_timeout_func) (const struct ticker_ext_timeout_data *data, void *context);
+
```

Comment:
That would work as well. I assume you mean that we can then use the same ticker_timeout_func callback function as before instead of having a different one; Extending the context is probably a bit cleaner code-wise

Reply:
yes. As we know the `ticker_timeout_func` was supplied using the extended interface hence the `context` is always `ticker_ext_context`.